File: frontend/src/app/features/tasks/models/task.dto.ts
export type TaskStatus = 'Suggested' | 'Open' | 'InProgress' | 'Done' | 'Cancelled';
export type TaskPriority = 'Low' | 'Normal' | 'High';
export type TaskOrigin = 'Manual' | 'AiSuggested' | 'AiApproved' | 'ImportedEmail' | 'ImportedFile';

export interface TaskListItemDto {
  id: string;
  title: string;
  dueDateUtc: string | null;
  status: TaskStatus;
  priority: TaskPriority;
  origin: TaskOrigin;
  assignedToFamilyMemberId: string | null;
  createdUtc: string;
}

export interface TaskDto extends TaskListItemDto {
  description: string | null;
  sourceDocumentId: string | null;
  createdByUserAccountId: string;
  isPrivate: boolean;
  updatedUtc: string;
  approvedByUserAccountId: string | null;
  approvedUtc: string | null;
  completedUtc: string | null;
}

export interface TaskListParams {
  status?: TaskStatus | undefined;
  assignedToFamilyMemberId?: string | undefined;
  priority?: TaskPriority | undefined;
  origin?: TaskOrigin | undefined;
  page?: number | undefined;
  pageSize?: number | undefined;
}

export interface CreateTaskRequest {
  title: string;
  description?: string | undefined;
  dueDateUtc?: string | undefined;
  priority: TaskPriority;
  assignedToFamilyMemberId?: string | undefined;
  isPrivate: boolean;
}

export interface PatchTaskRequest {
  title?: string | undefined;
  description?: string | undefined;
  dueDateUtc?: string | undefined;
  priority?: TaskPriority | undefined;
  assignedToFamilyMemberId?: string | undefined;
  isPrivate?: boolean | undefined;
}

File: src/FamilyOs.Application/Tasks/Dtos/TaskDto.cs
using FamilyOs.Domain.Enums;

namespace FamilyOs.Application.Tasks.Dtos;

public sealed class TaskDto
{
    public Guid Id { get; set; }
    public string Title { get; set; } = string.Empty;
    public string? Description { get; set; }
    public DateTime? DueDateUtc { get; set; }
    public string Status { get; set; } = string.Empty;
    public string Priority { get; set; } = string.Empty;
    public string Origin { get; set; } = string.Empty;
    public Guid? SourceDocumentId { get; set; }
    public Guid? AssignedToFamilyMemberId { get; set; }
    public Guid CreatedByUserAccountId { get; set; }
    public bool IsPrivate { get; set; }
    public DateTime CreatedUtc { get; set; }
    public DateTime UpdatedUtc { get; set; }
    public Guid? ApprovedByUserAccountId { get; set; }
    public DateTime? ApprovedUtc { get; set; }
    public DateTime? CompletedUtc { get; set; }
}

public sealed class TaskListItemDto
{
    public Guid Id { get; set; }
    public string Title { get; set; } = string.Empty;
    public string? Description { get; set; }
    public DateTime? DueDateUtc { get; set; }
    public string Status { get; set; } = string.Empty;
    public string Priority { get; set; } = string.Empty;
    public string Origin { get; set; } = string.Empty;
    public Guid? AssignedToFamilyMemberId { get; set; }
    public DateTime CreatedUtc { get; set; }
    public Guid? SourceDocumentId { get; set; }
    public string? SourceDocumentTitle { get; set; }
    public string? CardSummary { get; set; }
}
